src/Microsoft.Extensions.Configuration.AzureAppConfiguration/JsonConfiguration/JsonConfigurationParser.cs

```
@@ -0,0 +1,115 @@
+            Dictionary<string, string> keyValueDict = new Dictionary<string, string>();
+            ParseSetting(setting.Key, setting.Value, keyValueDict);
+            foreach (KeyValuePair<string, string> entry in keyValueDict)
+            {
+                keyValues.Add(entry);
+            }
+            return keyValues;
+        }
+
+
+        public static void ParseSetting(string currentKey, string currentValue, Dictionary<string, string> keyValueDict)
+        {
+            try
+            {
+                var json_data = JsonSerializer.Deserialize<JsonElement>(currentValue);
+                switch (json_data.ValueKind)
+                {
+                    case JsonValueKind.Array:
+                        var valueArray = json_data.EnumerateArray();
+                        for (int index = 0; index < valueArray.Count(); index++)
+                        {
+                            var newKey = ConfigurationPath.Combine(new List<string> { currentKey, index.ToString() });
+                            ParseSetting(newKey, valueArray.ElementAt(index).GetRawText(), keyValueDict);
+                        }
+                        break;
+
+                    case JsonValueKind.Object:
+                        var valueObject = json_data.EnumerateObject();
+                        foreach(JsonProperty entry in valueObject)
+                        {
+                            var newKey = ConfigurationPath.Combine(new List<string> { currentKey, entry.Name.ToString() });
+                            ParseSetting(newKey, entry.Value.GetRawText(), keyValueDict);
+                        }
+                        break;
+
+                    default:
+                        keyValueDict[currentKey] = json_data.ToString();
+                        break;
+                }
+
+            }
+            catch (Exception ex) when (ex is JsonException || ex is NotSupportedException)
```

>NotSupportedException [](start = 68, length = 21)

This might hide any deserialization bugs and cause silent failures. We might want to let it bubble up.

---

Done

---
In reply to: [454626098](https://github.com/Azure/AppConfiguration-DotnetProvider/pull/181#discussion_r454626098) [](ancestors = 454626098)